backend/src/controllers/weather_controller.ts

```diff
@@ -0,0 +1,158 @@
+import { Request, Response } from "express";
+import { start } from "repl";
+import { cities } from '../database/database';
+
+export const getCities = async (req: Request, res: Response) => {
+  try {
+    const data = cities.map(({ id, city, country, lat, lon }) => { return { id, city, country, longName: `${city} (${country})` } })
+    res.json(data).status(200);
+  } catch (error) {
+    res.status(400).json({ message: (error as Error).message });
+  }
+};
+
+
+export const getCityFiveDayForecast = async (req: Request, res: Response) => {
+  try {
+    console.log(req.query);
+    if (!req.query.id) {
+      res.status(404).json("City ID required.");
+    } else {
+
+      const city = cities.find(city => city.id === req.query.id);
+
+      if (!city) {
+        res.status(404).json("Unknown city.");
+      } else {
+
+        interface ApiDaily {
+          weather: {
+            icon: string;
+            main: string;
+            description: string;
+          }[];
+          rain?: number;
+          snow?: number;
+          uvi: number;
+          wind_deg: number;
+          wind_speed: number;
+          temp: {
+            day: number;
+            min: number;
+            max: number;
+          };
+          pop: number;
+          clouds: number;
+          sunrise: number;
+          sunset: number;
+          humidity: number;
+        }
```

Comment:
Can this interface be moved to the top of the file? Not sure what the 'correct' way is, but feels like this is nested deep in ifs and could be changed in the future.

Reply:
Could even be separated into its own file as there a few more underneath

Reply:
I'll take a look

Reply:
We will rearrange the code, sure. This is just work in progress.  I am sure there's room for refactoring too.

Reply:
But why not just merge for now - it works as is and we can worry about 'cosmetics' later.